File: fabric-audit-agent/adapters/lifecycle.store.js
import { readFile, writeFile, mkdir } from 'node:fs/promises';
import { dirname } from 'node:path';

/** Persists the lifecycle states map (key -> state record). Prod-DB at transfer. */
export function createLifecycleStore(filePath) {
  return {
    async load() {
      try { return JSON.parse(await readFile(filePath, 'utf-8')); }
      catch (err) { if (err.code === 'ENOENT') return {}; throw err; }
    },
    async save(states) {
      await mkdir(dirname(filePath), { recursive: true });
      await writeFile(filePath, JSON.stringify(states, null, 2), 'utf-8');
      return states;
    },
  };
}

File: fabric-audit-agent-app/fabric_audit_agent/adapters/lifecycle_store.py
"""Lifecycle-state StorePort (key -> state record). Port of ``adapters/lifecycle.store.js``.

Persists the per-finding lifecycle map (snooze/ack/resolve). Prod-DB at deploy.
"""
import json
import os


def create_lifecycle_store(file_path):
    def load():
        try:
            with open(file_path, "r", encoding="utf-8") as fh:
                return json.load(fh)
        except FileNotFoundError:
            return {}

    def save(states):
        os.makedirs(os.path.dirname(file_path) or ".", exist_ok=True)
        with open(file_path, "w", encoding="utf-8") as fh:
            json.dump(states, fh, indent=2, ensure_ascii=False)
        return states

    return {"load": load, "save": save}
